### ISSUE TITLE
Derive ZIO Test Gen[R, A] from Schema[A]

### ISSUE THREAD

**Issue by jdegoes** [zio/zio-schema#56]
Inside a new top-level project called `zio-schema-zio-test`, we can provide a derivation of ZIO Test's `Gen` for any data type that has a `Schema[A]`.

```scala
package zio.schema

object DeriveGen {
  def gen[A](implicit schema: Schema[A]): Gen[Any, A] = ???
}
```

**Comment by igrebenik** [comment]
I would like to claim this task for myself

**Comment by igrebenik** [comment]
@jdegoes John I started working on this task by describing Gen for primitives, then I thought about putting that together by traversing the whole Schema structure and I noticed that the folks working on the "Add a generic data representation" #36 did the same thing and did a lot more than I did! So maybe it makes sense for them to finish creating Gen instances as well to avoid double work...So it would just have to use certain Gen instances in that task to finish that task.

Or maybe I'm wrong and just didn't understand the task

**Comment by thinkharderdev** [comment]
Closed by #161
